### ISSUE TITLE
Set up loads of concepts (+maybe other trackers) to help ops keep on top of everything

### ISSUE THREAD

**Issue by plexish** [WikiApiary/WikiApiary#149]
Some may exist and just need connecting to a query on WikiApiary:Operations, some may not end up being very useful on further consideration, some I'll have missed (feel free to edit stuff in). Many will want a line or two of "what do I do to things on this list".
- [ ] Websites without URL
- [ ] Websites without API URL
- [x] Non-defunct active websites which have not been successfully collected from in one month
- [x] Websites using the default logo
- [ ] Untagged websites
- [ ] Websites missing tagline
- [ ] Websites missing description
- [ ] Websites audited but missing timestamp
- [ ] Generators lacking release dates (may need to exclude alpha and/or certain others, or find a way to deal with rolling alpha builds)
- [ ] Generators lacking release stability
- [ ] Generators lacking release type
- [ ] Generators lacking release status
- [ ] Extensions 100+ installs lacking description (?)
- [ ] Extensions ???+ lacking ohloh name (?)
- [ ] Skins 2+ default or 5+ installed lacking URL
- [ ] Skins 5+ default or 20+ installed lacking description
- [x] Farms with no successful checks on any sites in the last month
- [ ] Non-language farms with under 5 sites (worth checking to grab the rest of the farm)

Also related:
- [ ] Improve duplicate checker to account for protocol differences (http vs https)
- [ ] Improve duplicate checker to handle check for sites with same URL (ideally using some fun regex to strip out the specific page of the wiki, but may not be practical with my regex skills)

**Comment by jthingelstad** [comment]
Please consider Semantic MediaWiki concepts for most if not all of these. I can also generate them via cron and they can be nice and fast with a concept.

**Comment by plexish** [comment]
Okay, I'll use Concepts wherever possible :). I can also switch over most of the report box stuff to concepts if that would help performance?

**Comment by jthingelstad** [comment]
There is a big challenge with making these concepts in that Semantic MediaWiki currently lacks a way to search for pages that do not have a property. This is something I really wish we could do, but to my knowledge there is no way to do it.

**Comment by plexish** [comment]
Yea, that is an annoying issue for a lot of these and something I've run into a few times.. Possible workarounds include creating a default "none" or "unknown" value for the property where it would not interfere with bots, using {{#set: to add a new property such has Websites missing URL, or adding a hidden category and searching for pages in that category via concept. I'm not sure which is the best approach to take.

**Comment by kghbln** [comment]
Let's take the example "Websites without API URL". The only way to do it is to use a helper property of type boolean which checks for existence, e.g.

<pre>
{{#if:
 {{{API URL|}}}
 | {{#set: API URL provided = true }}
 | {{#set: API URL provided = false }}
}}
</pre>

and allows to query for a value (true or false) in any case.

I use this method sometimes and I think this is also the recommended approach (mailing list). However we will end up with a lot of extra properties.

**Comment by jthingelstad** [comment]
If we did this I would like to only set one condition. So rather than a true or false, just se a "Missing API URL" to true, and don't set a false condition.

**Comment by plexish** [comment]
I don't have a strong opinion about how it's done, it seems like there's four ways which would all be pretty much equally functional (true/false tracker property, mono-value tracker property, "not provided"/"unknown" default value for primary property, and category tracked by concept). The third could confuse the bots so is worst in many cases, but in places where I'm fairly certain it won't because the bots don't have reason to read/touch the values and the type of value is string it seems best.

For the ones which the third option won't work it'd be interesting to know which is the least resource intensive, but I'm happy with just using with the mono-value tracker property if thingles prefers, unless there's some advantage to double value or category I'm missing?

**Comment by kghbln** [comment]
The "not provided" / "unknown" will not work in this particular case since the property for the API is of type URL, but there are cases which very well will be ok with this approach. Just a matter of the respective case. Yeah, just setting a "false" is definitely fine. :)

What about setting some of them up as [fixed properties](https://semantic-mediawiki.org/wiki/Help:Fixed_properties)? I guess WikiAPIary will then be among the first wikis to do so. Probably there will be a couple of thousand values in some cases that will justify doing so. It should be fine since SESP is using the same approach and so far I have not heard worries about it. I personally have not tested this yet since I still on 1.8.

**Comment by jthingelstad** [comment]
Please don't put a "fake" value in the property to indicate it isn't there. I really am not a fan of that approach.

**Comment by plexish** [comment]
Okay, true/false tracker properties with concepts it is. Fixed properties look interesting, but somewhat experimental?

**Comment by jthingelstad** [comment]
Why not use the existing flags property and just store multiple flags in it?

I plan in setting up fixed properties. Now that SESP used them I have more confidence in them. This is partly why I want to reuse properties where we can. For example, we have three tag properties (websites, extensions and skins) and I want to merge them into a single property and them make it fixed.

**Comment by plexish** [comment]
That would work too, I'd not considered it. You'd prefer flags then?

And cool about fixed properties, if you want I can handle the tag merge, maybe tomorrow? I've used mass find/replace quite a lot on one of my wikis.

**Comment by jthingelstad** [comment]
No need to even find replace, I think it only requires some modest changes in the properties that are set in the templates.

**Comment by plexish** [comment]
Yep, I realized after I went to bed that find/replace was not necessary, was way too late. I'll take a look at the templates this evening :).

Edit: Done. All merged into Has tag (which was being used by websites already).

**Comment by plexish** [comment]
I've created [Has flag](https://wikiapiary.com/wiki/Property:Has_flag) and will use that with concepts to make most of these.

**Comment by plexish** [comment]
Did Farms with no recent collections, set a few farms to defunct.
